pages/partials/questionFooter.ejs

```diff
@@ -16,18 +16,27 @@
             </button>
           <% } %>
           <% if (showGradeButton) { %>
-              <% if (question_context == 'student_homework') { %>
-                  <button class="btn btn-primary question-grade disable-on-submit order-1 mr-1" name="__action" value="grade">Save & Grade
-                      <% if ((assessment_question.tries_per_variant - variant.num_tries) > 1) { %>
-                          <span class="badge">
-                              <%= String(assessment_question.tries_per_variant - variant.num_tries) %>
-                              tries left
-                          </span>
-                      <% } %>
-                  </button>
-              <% } else { %>
-                  <button class="btn btn-primary question-grade disable-on-submit order-1 mr-1" name="__action" value="grade">Save & Grade</button>
-              <% } %>
+            <button class="btn btn-primary question-grade disable-on-submit order-1 mr-1" name="__action" value="grade">Save & Grade
+            <%  let attemptsLeft = 0, attemptsTotal = 0;
+                if (question_context == 'student_homework') {
```

Comment:
I think the exam case works really well, but I actually think it's a bit confusing for homeworks. HW questions allow unlimited attempts, just at different variants. I think students will be confused when we say "single attempt" but then they can try the "same" question again (but with a different variant).

Maybe for homeworks we can do:
* if tries_per_variant > 1, then in the Grade&Submit button say "(2 attempts left at this version)" or "(Last attempt at this version)"
* for any homework question, put some extra text on the right side of the footer (not inside the Save&Grade button) that says something like "Unlimited question versions can be attempted"

That is, we won't display "Single attempt" for homework questions.

Reply:
May be a better idea to move the whole text regardless to outside of the button, using a badge instead? Having long text inside the button seems a bit crowded.

Reply:
I moved the badge outside the button. An alternative would be to move the badge back into the button and use badge-light to highlight it.

Reply:
I guess I think the clearest thing is to put the badge inside the button. Otherwise the student needs to make the association between the "Save&Grade" button and the text "XX grade attempts left", and they have to realize the badge doesn't apply to "Save", which all feels a bit complicated. Putting the badge inside the button makes it very obvious what's going on. I agree that the large button is a bit ugly, but I prefer clarity over beauty 😄

In terms of styling, I played around with some options below. I think that in isolation (like in this comment) option (2) looks best. But on the actual page the light badge inside the button looks a bit cluttered to me, and I think I actually prefer option (5) with parentheses, or possibly option (6). What do you think?

(1) `<span class="badge badge-light">`
<img width="233" alt="image" src="https://user-images.githubusercontent.com/875833/103138382-9ba1a480-4697-11eb-80e1-22e990b6562f.png">

(2) `<span class="badge badge-light ml-2">`
<img width="241" alt="image" src="https://user-images.githubusercontent.com/875833/103138406-f89d5a80-4697-11eb-9389-7675973c1196.png">

(3) `<span class="badge">`
<img width="243" alt="image" src="https://user-images.githubusercontent.com/875833/103138349-38177700-4697-11eb-9f30-476244f75634.png">

(4) `<small class="font-weight-bold ml-1">`
<img width="244" alt="image" src="https://user-images.githubusercontent.com/875833/103138783-89296a00-469b-11eb-8db9-4a557cd3b475.png">

(5) `<small class="ml-1">`
<img width="235" alt="image" src="https://user-images.githubusercontent.com/875833/103138699-bfb2b500-469a-11eb-82a0-ace02003519d.png">

(6) `<small class="font-italic ml-2">`
<img width="230" alt="image" src="https://user-images.githubusercontent.com/875833/103138906-9135d980-469c-11eb-826b-a292e51b2923.png">

Reply:
Option 6 actually looks pretty good.

Reply:
And I still think that we should add something like `<span class="d-flex align-self-center"> Unlimited question versions can be attempted</span>` to the end of the footer for homework questions. This will help us with #3347, where we want to add the ability to mix and match question grading schemes. It would be great if the question footer could clearly indicate both:
1. How many attempts are left at this variant.
2. Are new variants going to be allowed?

Reply:
> Option 6 actually looks pretty good.

Yeah, I'm definitely not opposed to that.

Reply:
More playing with styling: For homework questions with unlimited variants I like:

`<span class="d-flex align-self-center"><small class="font-italic">Unlimited question versions can be attempted<small></small></small></span>`
<img width="832" alt="image" src="https://user-images.githubusercontent.com/875833/103139172-2b971c80-469f-11eb-8d2b-0d7b6418a2e9.png">

Thinking about getting ready for #3347 I propose:
1. The button text always shows how many attempts are remaining for the current variant (e.g., "6 attempts left", "Last attempt", "Single attempt", "unlimited attempts", "blocked for 5 min").
2. We add extra text like above to the right-hand side of the footer to explain whether new variants will be allowed. If new variants are not allowed (like on exams or for single-variant questions) then we don't say anything. If new variants are allowed then we say something.

What do you think about this?

I'm not sure whether we should use the work "variant" or "version" (or something else?) to refer to the thing that we internally call a `variant`. I generally prefer to keep internal and external terminology the same whenever possible, but sometimes it's clearer to use different words. What do you think?

In any case, maybe we should change the button that says "Try question again" to "Try another question variant" or something like that?

Reply:
> And I still think that we should add something like <span class="d-flex align-self-center"> Unlimited question versions can be attempted</span> to the end of the footer for homework questions. 

I'm okay with that, just not for single variant questions.

Reply:
> `<span class="d-flex align-self-center"><small class="font-italic">Unlimited question versions can be attempted<small></small></small></span>`

Wording sounds weird in juxtaposition with the button text. Maybe "Additional attempts available with different variants"? Just to make it clear the difference between attempts in the button and attempts in other variants.

Reply:
> I'm okay with that, just not for single variant questions.

Yep, absolutely, we already have the distinction between "new variants allowed" and "no new variants" (singleVariant) for homeworks. It would be great if the question footer made this clear.

Reply:
> Wording sounds weird in juxtaposition with the button text. Maybe "Additional attempts available with different variants"? Just to make it clear the difference between attempts in the button and attempts in other variants.

I like it! Maybe "... with new variants", and we could change "Try question again" to "Try a new question variant" to link together the ideas? Or use "different" rather than "new" in both cases.

We could also add a question-mark icon next to "Additional attempts available..." with a popover that more clearly explains what variants are and the idea of trying new variants. This is a somewhat unusual feature of PL, so it wouldn't hurt to explain it more for students. Although it seems like they mainly get the idea very quickly when they just use the system.